app/controllers/chatfuel_controller.rb

```diff
@@ -1,12 +1,16 @@
 class ChatfuelController < ApplicationController
   def show
-    chatbot_channel = Channel.chatbot
+    @topic = Topic.find_by(name: params[:topic])
 
-    @text_component = Text::Sorter.sort(chatbot_channel.text_components, {}).first
+    if @topic
+      @text_component = Text::Sorter.sort(@topic.text_components, {}).first
```

Comment:
What about text components for the topic but which are not assigned to the chatbot channel? It feels horribly over-engineered for me, but configuring every text component explicitly to go into the chatbot was required by our editorial team, @drjakob is that correct?